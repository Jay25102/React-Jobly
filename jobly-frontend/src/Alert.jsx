import React from "react";

/**
 * Component to display alerts in forms
 */
function Alert({type = "danger", messages = []}) {
    console.debug("Alert", "type=", type, "messages=", messages);

    return (
        <div role="alert">
            {messages.map(error => (
                <p key={error}>
                    {error}
                </p>
            ))}
        </div>
    )
}

export default Alert;