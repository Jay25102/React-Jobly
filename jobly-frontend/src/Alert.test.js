import React from "react";
import { render } from "@testing-library/react";
import Alert from "./Alert";

describe("Alert", function() {
    test("renders without crashing", function() {
        render(<Alert/>);
    });
    test("matches snapshot for danger messages", function() {
        let messages = ["this broke", "please help"];
        const { asFragment } = render(<Alert type="danger" messages={messages}/>);
        expect(asFragment()).toMatchSnapshot();
    });
    test("matches snapshot for success messages", function() {
        let messages = ["this works", "yay"];
        const { asFragment } = render(<Alert type="success" messages={messages}/>);
        expect(asFragment()).toMatchSnapshot();
    });
})