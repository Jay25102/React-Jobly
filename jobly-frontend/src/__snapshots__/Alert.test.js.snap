// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert matches snapshot for danger messages 1`] = `
<DocumentFragment>
  <div
    role="alert"
  >
    <p>
      this broke
    </p>
    <p>
      please help
    </p>
  </div>
</DocumentFragment>
`;

exports[`Alert matches snapshot for success messages 1`] = `
<DocumentFragment>
  <div
    role="alert"
  >
    <p>
      this works
    </p>
    <p>
      yay
    </p>
  </div>
</DocumentFragment>
`;
